test/connectionProfile.test.ts

```diff
@@ -294,22 +294,22 @@ suite('Connection Profile tests', () => {
 
 		let vscodeWrapperMock = TypeMoq.Mock.ofType(VscodeWrapper);
 		vscodeWrapperMock.setup(x => x.activeTextEditorUri).returns(() => 'test.sql');
+		// user cancels out of error message
 		vscodeWrapperMock.setup(x => x.showErrorMessage(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(undefined));
 
 		let connectionUI = new ConnectionUI(connectionManagerMock.object, contextMock.object,
 			connectionStoreMock.object, mockAccountStore, prompter.object, vscodeWrapperMock.object);
 
 		// create a new connection profile
-		connectionUI.createAndSaveProfile().then(profile => {
+		connectionUI.createAndSaveProfile()
+		// CancelError will be thrown (expected)
+		.catch(() => {
```

Comment:
Why are you moving this into the catch? Isn't that changing the behavior of the test? Before if an exception was thrown it would automatically fail the test.

Reply:
The logic where this is run was changed in the linked PR, but here is the specific line where instead of undefined we throw the CancelError: https://github.com/microsoft/vscode-mssql/pull/17350/files#diff-ddbe2fb5d84649cfa54b0ed716fb17faf46f59b3d6a152115f32ee2c3b1cd227R571